testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/miniapplist/MiniAppListFragment.kt

```diff
@@ -62,35 +69,97 @@ class MiniAppListFragment : BaseFragment(), MiniAppList {
         viewModel = ViewModelProvider.NewInstanceFactory().create(MiniAppListViewModel::class.java).apply {
             miniAppListData.observe(viewLifecycleOwner, Observer {
                 swipeRefreshLayout.isRefreshing = false
-                displayMiniAppList(it)
+                addMiniAppList(it)
                 MiniAppListStore.instance.saveMiniAppList(it)
             })
             errorData.observe(viewLifecycleOwner, Observer {
                 val list = MiniAppListStore.instance.getMiniAppList()
                 if (list.isEmpty())
-                    Toast.makeText(context, it, Toast.LENGTH_LONG).show()
+                    updateEmptyView(list)
                 else
-                    displayMiniAppList(list)
-                swipeRefreshLayout.isRefreshing = false
+                    addMiniAppList(list)
+                    swipeRefreshLayout.isRefreshing = false
             })
         }
 
-        swipeRefreshLayout.setOnRefreshListener { executeLoadingList() }
+        swipeRefreshLayout.setOnRefreshListener {
+            executeLoadingList()
+            resetSearchBox()
+        }
+    }
+
+    private fun addMiniAppList(list: List<MiniAppInfo>) {
+        downloadedList = list
+        updateMiniAppListState(list)
     }
 
-    private fun displayMiniAppList(list: List<MiniAppInfo>) {
+    private fun updateMiniAppListState(list: List<MiniAppInfo>) {
         miniAppListAdapter.addListWithSection(list)
+        updateEmptyView(list)
     }
 
     private fun executeLoadingList() {
         viewModel.getMiniAppList()
     }
 
     override fun onMiniAppItemClick(miniAppInfo: MiniAppInfo) {
-        raceExecutor.run { MiniAppDisplayActivity.start(context!!, miniAppInfo) }
+        resetSearchBox()
+        raceExecutor.run {
+            context?.let { MiniAppDisplayActivity.start(it, miniAppInfo) }
+        }
     }
 
     fun switchToInput() {
         raceExecutor.run { activity?.launchActivity<MiniAppInputActivity>() }
     }
+
+    override fun onPrepareOptionsMenu(menu: Menu) {
+        super.onPrepareOptionsMenu(menu)
+        addSearchBox(menu)
+    }
+
+    private fun addSearchBox(menu: Menu) {
+        val itemSearch = menu.findItem(R.id.action_search)
+        searchView = itemSearch.actionView as SearchView
+        searchView.queryHint = getString(R.string.menu_hint_search_miniapps)
+        searchView.setOnQueryTextListener(this)
```

Comment:
This is a potential leak, the listener is being added every time. IIRC, this is called every time whereas, onCreateOptionsMenu happens once. 
https://developer.android.com/guide/topics/search/search-dialog

Reply:
I have updated this part and now initializing the search menu and it's listener using `onCreateOptionsMenu`. So, it's calling only once.